sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -185,19 +216,44 @@ class PrefixKeyScanStateEncoder(
     restoreKeyProjection(joinedRowOnKey.withLeft(prefixKeyDecoded).withRight(remainingKeyDecoded))
   }
 
+  override def decodeKey(
+      keyBytes: Array[Byte],
+      hasVirtualColFamilyPrefix: Boolean = false): UnsafeRow = {
+    decodeKeyWithColFamilyPrefix(keyBytes, hasVirtualColFamilyPrefix)
+  }
+
   private def extractPrefixKey(key: UnsafeRow): UnsafeRow = {
     prefixKeyProjection(key)
   }
 
-  override def encodePrefixKey(prefixKey: UnsafeRow): Array[Byte] = {
+  private def encodePrefixKeyWithColFamilyPrefix(
+    prefixKey: UnsafeRow,
+    hasVirtualColFamilyPrefix: Boolean = false,
+    colFamilyId: Long = -1L): Array[Byte] = {
+    val offSetForColFamilyPrefix = if (hasVirtualColFamilyPrefix) 8 else 0
+
     val prefixKeyEncoded = encodeUnsafeRow(prefixKey)
-    val prefix = new Array[Byte](prefixKeyEncoded.length + 4)
-    Platform.putInt(prefix, Platform.BYTE_ARRAY_OFFSET, prefixKeyEncoded.length)
+    val prefix = new Array[Byte](
+      prefixKeyEncoded.length + 4 + offSetForColFamilyPrefix)
+    if (hasVirtualColFamilyPrefix) {
+      Platform.putLong(prefix, Platform.BYTE_ARRAY_OFFSET, colFamilyId)
```

Comment:
`Short` might be enough ? 2 bytes probably good enough for 0 indexed num of column families ?

Reply:
Use `Short` instead of `Long` as VCF id.